cmd/aad-cli/cli/user_test.go

```diff
@@ -134,13 +135,20 @@ func TestUserSetAttribute(t *testing.T) {
 		tc := tc
 		t.Run(name, func(t *testing.T) {
 			args := strings.Split(tc.args, " ")
-			username := args[slices.Index(args, "--name")+1]
+
+			usernameIndex := slices.Index(args, "--name")
+			username := args[usernameIndex+1]
+
+			// Fallback when a username is not provided
+			if usernameIndex == -1 {
+				username = "futureuser@domain.com"
+			}
 
 			cacheDir := t.TempDir()
 			cacheDB := "db_with_old_users"
 			testutils.PrepareDBsForTests(t, cacheDir, cacheDB)
 			cache := testutils.NewCacheForTests(t, cacheDir)
-			c := cli.New(cli.WithCache(cache))
+			c := cli.New(cli.WithCache(cache), cli.WithCurrentUser("futureuser@domain.com"))
```

Comment:
Perfect! This is the way to go to allow this use case (there is unfortunately no way in the integration tests to change what `user.Current()` returns, as it’s a syscal. Yes, not the first time we ecounter this!